### ISSUE TITLE
disallowed methods unreachable warning doesn't fail with -D warnings

### ISSUE THREAD

**Issue by ijackson** [rust-lang/rust-clippy#17007]
### Summary

The warning "does not refer to a reachable function" does not cause a nonzero exit status  even with `-D warnings`

### Steps

```
git clone https://gitlab.torproject.org/tpo/core/arti
cd arti
git checkout f7b6a9d27408d07fafce9c90530177abe7b751ab
cargo clippy --locked --all-features --workspace --all-targets -- -D warnings
echo $?
```

### Expected results with #17006 unfixed

```
warning: `rand::RngExt::random_range` does not refer to a reachable function
 --> /volatile/rustcargo/d/arti/clippy.toml:7:5
  |
7 |     { path = "rand::RngExt::r...fallible if applicable" },
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^...^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: add `allow-invalid = true` to the entry to suppress this warning
 
warning: `tor-geoip` (lib test) generated 1 warning
```

Plus a nonzero exit status.

### Actual results

```
warning: `rand::RngExt::random_range` does not refer to a reachable function
 --> /volatile/rustcargo/d/arti/clippy.toml:7:5
  |
7 |     { path = "rand::RngExt::r...fallible if applicable" },
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^...^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: add `allow-invalid = true` to the entry to suppress this warning
 
warning: `tor-geoip` (lib test) generated 1 warning
...
0
```

### Notes

After #17006 is fixed, this repro won't work any more.  Presumably an actually-unknown function would cause the same issue, but I haven't checked.

### Version

```text
rustc 1.96.0-beta.5 (a5a9a5438 2026-05-01)
binary: rustc
commit-hash: a5a9a5438ba8f6b83523031de99b15c12b44cea4
commit-date: 2026-05-01
host: x86_64-unknown-linux-gnu
release: 1.96.0-beta.5
LLVM version: 22.1.2
```
